packages/api/src/command/medical/patient/consolidated-get.ts

```diff
@@ -110,7 +130,68 @@ export async function startConsolidatedQuery({
   return progress;
 }
 
-async function getConsolidatedAndSendToCx(params: GetConsolidatedParams): Promise<void> {
+function appendProgressToProcessingQueries(
+  currentConsolidatedQueries: ConsolidatedQuery[] | undefined,
+  progress: ConsolidatedQuery
+): ConsolidatedQuery[] {
+  if (currentConsolidatedQueries) {
+    const queriesInProgress = currentConsolidatedQueries.filter(
+      query => query.status === "processing"
+    );
+
+    return [...queriesInProgress, progress];
+  }
+
+  return [progress];
+}
+
+export function getCurrentConsolidatedProgress(
+  consolidatedQueriesProgress: ConsolidatedQuery[] | undefined,
+  queryParams: GetConsolidatedFilters
+): ConsolidatedQuery | undefined {
+  if (!consolidatedQueriesProgress) return undefined;
+
+  const { resources, dateFrom, dateTo, conversionType } = queryParams;
+
+  for (const progress of consolidatedQueriesProgress) {
+    const isSameResources = getIsSameResources(resources, progress.resources);
+    const isSameDateFrom = progress.dateFrom === dateFrom;
+    const isSameDateTo = progress.dateTo === dateTo;
+    const isSameConversionType = progress.conversionType === conversionType;
+    const isProcessing = progress.status === "processing";
+
+    if (isSameResources && isSameDateFrom && isSameDateTo && isSameConversionType && isProcessing) {
+      return progress;
+    }
+  }
+}
+
+export function getIsSameResources(
+  queryResources: ResourceTypeForConsolidation[] | undefined,
+  currentResources: ResourceTypeForConsolidation[] | undefined
+): boolean {
+  const haveSameLength = queryResources?.length === currentResources?.length;
```

Comment:
It seems we don't need this check, as we're already looking at the intersection of the arrays, no?

Reply:
Actually, using it introduces a bug when the length is the same but the contents are different (considering that my understanding is correct, that we only consider the same if the list of resources is also the same).

Reply:
We need the length because `const resp = getIsSameResources([], ["Observation"]);` will be true because 

queryResources = 0
intersection = 0

Reply:
I see the problem you reported, and I agree we need to account for that.

But what about returning `true` when the intersection is different but the lengths are the same?

For the issue you reported, we could check if `queryResources` empty or has zero items - then its not an intersection.